postgres/datadog_checks/postgres/postgres.py

```diff
@@ -180,10 +181,39 @@ def _new_query_executor(self, queries):
         )
 
     def execute_query_raw(self, query):
-        with self._get_main_db().cursor() as cursor:
-            cursor.execute(query)
-            rows = cursor.fetchall()
-            return rows
+        with self.db() as conn:
+            with conn.cursor() as cursor:
+                cursor.execute(query)
+                rows = cursor.fetchall()
+                return rows
+
+    @contextlib.contextmanager
+    def db(self):
```

Comment:
This is the change that takes care of the premature connection close. Instead of a persisted `self.db` property, we will use a `db` context manager to make sure a healthy connection is yield. If the connection is closed, we will reinitialize the connection